rclc/test/rclc/test_executor.cpp

```diff
@@ -1989,6 +2006,106 @@ TEST_F(TestDefaultExecutor, executor_test_service_with_reqid) {
+
+  rc = rclc_executor_add_service_with_context(
+    &executor, &service,
+    &req, &resp,
+    &service_callback_with_context,
+    &srv1_id);
+  EXPECT_EQ(RCL_RET_OK, rc) << rcl_get_error_string().str;
+  EXPECT_EQ(executor.info.number_of_clients, (size_t) 0);
+  EXPECT_EQ(executor.info.number_of_services, (size_t) 1);
+
+  // Creating client and options
+  rcl_client_options_t client_options = rcl_client_get_default_options();
+  rcl_client_t client = rcl_get_zero_initialized_client();
+  rc = rcl_client_init(&client, &this->node, service_type_support, service_name, &client_options);
+  EXPECT_EQ(RCL_RET_OK, rc) << rcl_get_error_string().str;
+
+  // client messages
+  example_interfaces__srv__AddTwoInts_Request cli_req;
+  example_interfaces__srv__AddTwoInts_Request__init(&cli_req);
+  example_interfaces__srv__AddTwoInts_Response cli_resp;
+  example_interfaces__srv__AddTwoInts_Response__init(&cli_resp);
+
+  // add client to executor
+  rc = rclc_executor_add_client(
+    &executor, &client, &cli_resp,
+    client_callback);
+  EXPECT_EQ(RCL_RET_OK, rc) << rcl_get_error_string().str;
+  EXPECT_EQ(executor.info.number_of_clients, (size_t) 1);
+  EXPECT_EQ(executor.info.number_of_services, (size_t) 1);
+
+  // send client request
+  int64_t seq;
+  cli_req.a = 1;
+  cli_req.b = 2;
+  rc = rcl_send_request(&client, &cli_req, &seq);
+  EXPECT_EQ(RCL_RET_OK, rc) << rcl_get_error_string().str;
+  EXPECT_EQ(seq, (int64_t) 1);  // sequence id = 1
+
+  // initialize test results
+  _results_initialize_service_client();
+  EXPECT_EQ(srv1_cnt, (unsigned int) 0);
+  EXPECT_EQ(srv1_value, (unsigned int) 0);
+  EXPECT_EQ(srv1_id, (unsigned int) 0);
+
+  // spin executor, which will
+  // - receive request from client
+  // - call service_callback function
+  // - send response message to client
+  std::this_thread::sleep_for(rclc_test_sleep_time);
+  rclc_executor_spin_some(&executor, rclc_test_timeout_ns);
+
+  EXPECT_EQ(srv1_cnt, (unsigned int) 1);  // check that service callback was called
+  EXPECT_EQ(srv1_value, (unsigned int) 1);  // check value of 'a' in request message
+  EXPECT_EQ(srv1_id, (unsigned int) 1);  // check sequence id
+
+  // spin executor, which will
+  // - receive response message from server
+  // - call client_callback
+  std::this_thread::sleep_for(rclc_test_sleep_time);
+  rclc_executor_spin_some(&executor, rclc_test_timeout_ns);
+
+  EXPECT_EQ(cli1_cnt, (unsigned int) 1);  // check that client callback was called
+  EXPECT_EQ(cli1_value, (unsigned int) 3);  // check value of 'sum' in response message
+  EXPECT_EQ(cli1_id, (unsigned int) 1);  // check sequence id
+
+  // tear down
+  rc = rcl_service_fini(&service, &this->node);
+  EXPECT_EQ(RCL_RET_OK, rc) << rcl_get_error_string().str;
+  rc = rcl_client_fini(&client, &this->node);
+  EXPECT_EQ(RCL_RET_OK, rc) << rcl_get_error_string().str;
+  rc = rclc_executor_fini(&executor);
+  EXPECT_EQ(RCL_RET_OK, rc) << rcl_get_error_string().str;
+}
```

Comment:
just to be sure call fini method of the messages as well.
here probably this is not needed because you have only one unit test, but if you add more unit tests, the "space is clean again".

example_interfaces__srv__AddTwoInts_Request__fini(&req);
example_interfaces__srv__AddTwoInts_Response__fini(&resp);
example_interfaces__srv__AddTwoInts_Request__fini(&cli_req);
example_interfaces__srv__AddTwoInts_Response__fini(&cli_resp);